mapit/views/areas.py

```diff
@@ -394,8 +394,15 @@ def areas_by_point(request, srid, x, y, bb=False, format=''):
 
     q = query_args(request, format)
 
+    try:
+        within = float(request.GET.get('within', 0))
+    except ValueError:
+        raise ViewException(format, _('Bad within specified'), 400)
```

Comment:
This error is a bit terse and hard to parse, something like `'Bad "within" parameter specified'` would be clearer.